src/sysupdate/sysupdated.c

```diff
@@ -0,0 +1,1914 @@
+                "name", t->name,
+                NULL
+        };
+
+        r = bus_verify_polkit_async(
+                msg,
+                "org.freedesktop.sysupdate1.vacuum",
+                details,
+                &t->manager->polkit_registry,
+                error);
+        if (r < 0)
+                return r;
+        if (r == 0)
+                return 1; /* Will call us back */
+
+        r = job_new(JOB_VACUUM, t, msg, target_method_vacuum_finish, NULL, &j);
+        if (r < 0)
+                return r;
+
+        r = job_start(j);
+        if (r < 0)
+                return sd_bus_error_set_errnof(error, r, "Failed to start job: %m");
+        TAKE_PTR(j); /* Avoid job from being killed & freed */
+
+        return 1;
+}
+
+static int target_method_get_version(sd_bus_message *msg, void *userdata, sd_bus_error *error) {
+        Target *t = ASSERT_PTR(userdata);
+        _cleanup_free_ char *target_arg = NULL;
+        _cleanup_(sd_json_variant_unrefp) sd_json_variant *v = NULL;
+        sd_json_variant *version_json;
+        int r;
+
+        r = target_get_argument(t, &target_arg);
+        if (r < 0)
+                return r;
+
+        r = sysupdate_run_simple(&v, "--offline", "list", target_arg, NULL);
+        if (r < 0)
+                return r;
+
+        version_json = sd_json_variant_by_key(v, "current");
+        if (!version_json) {
+                log_sysupdate_bad_json("list", "Missing key 'current'");
+                return -EINVAL;
+        }
+
+        if (sd_json_variant_is_null(version_json))
+                return sd_bus_reply_method_return(msg, "s", "");
+
+        if (!sd_json_variant_is_string(version_json)) {
+                log_sysupdate_bad_json("list", "Expected string value for key 'current'");
+                return -EINVAL;
+        }
+
+        return sd_bus_reply_method_return(msg, "s", sd_json_variant_string(version_json));
+}
+
+static int target_get_appstream(Target *t, char ***ret) {
+        _cleanup_free_ char *target_arg = NULL;
+        _cleanup_(sd_json_variant_unrefp) sd_json_variant *v = NULL;
+        sd_json_variant *appstream_url_json;
+        int r;
+
+        r = target_get_argument(t, &target_arg);
+        if (r < 0)
+                return r;
+
+        r = sysupdate_run_simple(&v, "--offline", "list", target_arg, NULL);
+        if (r < 0)
+                return r;
+
+        appstream_url_json = sd_json_variant_by_key(v, "appstream_urls");
+        if (!appstream_url_json) {
+                log_sysupdate_bad_json("list", "Missing key 'appstream_urls'");
+                return -EINVAL;
+        }
+
+        r = sd_json_variant_strv(appstream_url_json, ret);
+        if (r < 0) {
+                log_sysupdate_bad_json("list", "Expected array of strings for key 'appstream_urls'");
+                log_debug("Invalid JSON response from 'systemd-sysupdate list': Expected array of strings for key 'appstream_urls'");
```

Comment:
```
return log_debug_errno(r, "…");
```

Reply:
I've kept as `log_sysupdate_bad_json(...); return r;` so as to keep using the macro to generate the error message.